parser/src/lexer.rs

```diff
@@ -641,81 +722,117 @@ mod tests {
-            vec![
-                Tok::Def,
-                Tok::Name {
-                    name: String::from("foo"),
-                },
-                Tok::Lpar,
-                Tok::Rpar,
-                Tok::Colon,
-                Tok::Newline,
-                Tok::Indent,
-                Tok::If,
-                Tok::Name {
-                    name: String::from("x"),
-                },
-                Tok::Colon,
-                Tok::Newline,
-                Tok::Indent,
-                Tok::Return,
-                Tok::Number { value: 99 },
-                Tok::Newline,
-                Tok::Dedent,
-                Tok::Dedent,
-            ]
-        );
+    test_indentation_with_eol! {
+        test_indentation_windows_eol: WINDOWS_EOL,
+        test_indentation_mac_eol: MAC_EOL,
+        test_indentation_unix_eol: UNIX_EOL,
     }
 
-    #[test]
-    fn test_newline_in_brackets() {
-        let source = String::from("x = [\n    1,2\n]\n");
-        let tokens = lex_source(&source);
-        assert_eq!(
-            tokens,
-            vec![
-                Tok::Name {
-                    name: String::from("x"),
-                },
-                Tok::Equal,
-                Tok::Lsqb,
-                Tok::Number { value: 1 },
-                Tok::Comma,
-                Tok::Number { value: 2 },
-                Tok::Rsqb,
-                Tok::Newline,
-            ]
-        );
+    macro_rules! test_double_dedent_with_eol {
+        ($($name:ident: $eol:expr,)*) => {
+        $(
+            #[test]
+            fn $name() {
+                let source = String::from(format!("def foo():{} if x:{}{}  return 99{}{}", $eol, $eol, $eol, $eol, $eol));
+                let tokens = lex_source(&source);
+                assert_eq!(
+                    tokens,
+                    vec![
+                        Tok::Def,
+                        Tok::Name {
+                            name: String::from("foo"),
+                        },
+                        Tok::Lpar,
+                        Tok::Rpar,
+                        Tok::Colon,
+                        Tok::Newline,
+                        Tok::Indent,
+                        Tok::If,
+                        Tok::Name {
+                            name: String::from("x"),
+                        },
+                        Tok::Colon,
+                        Tok::Newline,
+                        Tok::Indent,
+                        Tok::Return,
+                        Tok::Number { value: 99 },
+                        Tok::Newline,
+                        Tok::Dedent,
+                        Tok::Dedent,
+                    ]
+                );
+            }
+        )*
+        }
+    }
+
+    test_double_dedent_with_eol! {
+        test_double_dedent_windows_eol: WINDOWS_EOL,
+        test_double_dedent_mac_eol: MAC_EOL,
+        test_double_dedent_unix_eol: UNIX_EOL,
+    }
+
+    macro_rules! test_newline_in_brackets {
```

Comment:
These macro's make the tests somewhat harder to understand in my opinion, but maybe this can be revised later on.

Reply:
What would you suggest? Call the test function three times with three different newlines as argument? And put these three calls into three separate test functions (for better identification) ?